contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,628 @@
+
+        _checkLiquidityWithinExpectedShare();
+    }
+
+    /**
+     * @dev Perform a swap so that after the swap the ticker has the desired WETH to OETHb token share.
+     */
+    function _addLiquidity() internal {
+        uint256 wethBalance = IERC20(WETH).balanceOf(address(this));
+        require(wethBalance > 0, "Must add some WETH");
+
+        // TODO mint corresponding 
+        //IVault(vaultAddress).mintForStrategy(oethBRequired);
+
+        if (tokenId == 0) {
+            (uint160 sqrtRatioX96, , , , ,) = clPool.slot0();
+            console.log("The current square root balance of the pool");
+            console.log(sqrtRatioX96);
+
+            // TODO add new token id position minted event
+            (uint256 mintedTokenId,,uint256 amountWETH, uint256 amountOETHb) = positionManager.mint(
+                INonfungiblePositionManager.MintParams({
+                    token0: WETH,
+                    token1: OETHb,
+                    tickSpacing: 1,
+                    tickLower: 0,
+                    tickUpper: 1,
+                    amount0Desired: wethBalance,
+                    amount1Desired: 0,
+                    amount0Min: 0,
+                    amount1Min: 0,
+                    recipient: address(this),
+                    deadline: block.timestamp,
+                    // needs to be 0 because the pool is already created
+                    // non zero amount attempts to create a new instance of the pool
+                    sqrtPriceX96: 0 
+                })
+            );
+            tokenId = mintedTokenId;
+            // TODO add incerase liquidity event
+            netValue += amountWETH + amountOETHb;
+        } else {
+
+            (,uint256 amountWETH, uint256 amountOETHb) = positionManager.increaseLiquidity(
+                INonfungiblePositionManager.IncreaseLiquidityParams({
+                    tokenId: tokenId,
+                    amount0Desired: wethBalance,
+                    amount1Desired: 0,
+                    amount0Min: 0,
+                    amount1Min: 0,
+                    deadline: block.timestamp
+                })
+            );
+            // TODO add incerase liquidity event
+            netValue += amountWETH + amountOETHb;
+        }
+
+        clGauge.deposit(tokenId);
+    }
+
+    /**
+     * @notice Deposit all supported assets in this strategy contract to the platform
+     */
+    function depositAll() external virtual override {
+        _deposit(WETH, IERC20(WETH).balanceOf(address(this)));
+    }
+
+    /**
+     * @notice Withdraw an `amount` of assets from the platform and
+     * send to the `_recipient`.
+     * @param _recipient         Address to which the asset should be sent
+     * @param _asset             Address of the asset
+     * @param _amount            Units of asset to withdraw
+     */
+    function withdraw(
+        address _recipient,
+        address _asset,
+        uint256 _amount
+    ) external virtual override {
```

Comment:
Missing implementation? (and authorization)

Reply:
Maybe withdraw is just a rebalance, but only adding X amount of liquidity back at the end.

Reply:
Does withdraw, no swap, take the WETH, then add liquidity back work?